### PR TITLE
fill in questions in functions.ts

### PR DESCRIPTION
**Pull request by vuhungthang**
Pass all tests, but please check number 4

### CODE REVIEW

File: exercises/basics/functions.ts [hunk 1]
```diff
@@ -1,25 +1,44 @@
...
 }
-
 export function curryConcept1(input : number, callback : (input2 : number) => number) {
-  // @TODO impl
-  //       you must combine `input` and `callback`
+    //return callback(3*2)+input // This one is cheating, however, it still passes, LMAO
+    const result = function(input2:number) {
+        return callback(input2) + input*2;
+    }
+    return result(input)
```
Comment (mattlo) [inline exercises/basics/functions.ts:21]: you solved this one just fine, you just did a superfluous step

you could have simply just did `return callback(input) + (input * 2)`;

File: exercises/basics/functions.ts [hunk 2]
```diff
@@ -1,25 +1,44 @@
...
+// This one is cheating,too but I will ask you later!
 }
-
 export function returnNestedFunction10LevelsDeep() {
-  // @TODO impl
-  // @TODO in typescript, you'll need to set the parameter type as `number`
-  //       otherwise the test will complain
-  //       (hello : number) => {}
+    return function(lv1:1) {
```
Comment (mattlo) [inline exercises/basics/functions.ts:25]: you answered this one correctly, but I'd like to see a change here:

1. use arrow functional expressions instead without the use of `return`
2. when you define the type `lv1: 1`, although that will pass in TypeScript, it really should be `lv1: number`... fix that for the rest

**Review by mattlo — CHANGES_REQUESTED** [review]
@vuhungthang I'm actually surprised you got this after that small lesson yesterday. I thought you would have more questions. Great job passing all the tests!

Some adjustments here to make it better.

File: exercises/basics/functions.ts (hunk 1 above)
Comment (vuhungthang) [reply exercises/basics/functions.ts]: I updated it! :>

File: exercises/basics/functions.ts (hunk 2 above)
Comment (vuhungthang) [reply exercises/basics/functions.ts]: I changed it!

**Review by mattlo — APPROVED** [review]

**Pull request merged**